Node API/routes.txt

```diff
@@ -0,0 +1,160 @@
+-----------------------------------------------Movies-----------------------------------------------
+
+
+--List movies (supporting paging, filtering and sorting)--
+
+HTTP Method: GET
+HTTP Path: /movies?limit=[limit]&offset=[offset]&filter=[filter]&sortField=[sortField]&sortDirection=[sortDirection]
+Path Parameters: limit=[integer], offset=[integer], filter=[string], sortField=[string], sortDirection=[string]
+Query parameters: None
+HTTP response codes: 200, 400, 404, 500
+Success response schema: [
+    { id, title, releaseDate, duration, actors, genres, director, description, rating, poster, trailer, createdAt, updatedAt }
+]
+Error response schema: { errorMessage, errorCode }
+
+--Get Movie--
+
+HTTP Method: GET
+HTTP Path: /movies/[movieId]
+Path Parameters: movieId=[integer]
+Query parameters: None
+HTTP response codes: 200, 400, 404, 500
+Success response schema: { id, title, releaseDate, duration, actors, genres, director, description, rating, poster, trailer, createdAt, updatedAt }
+Error response schema: { errorMessage, errorCode }
+
+
+-----------------------------------------------Lists-----------------------------------------------
+
+
+--Get user lists--
+
+HTTP Method: GET
+HTTP Path: /users/[userId]/lists
+Path Parameters: userId=[integer]
+Query parameters: None
+HTTP response codes: 200, 400, 404, 500
+Success response schema: [
+    { id, name, description, movies, createdAt, updatedAt }
+]
+Error response schema: { errorMessage, errorCode }
+
+--Create List--
+
+HTTP Method: POST
+HTTP Path: /users/[userId]/lists/[listId]
```

Comment:
The path sould be based on `lists` instead of `users`. Alse the `listId` is a mistake here, since we don't have the `id` of the resource we want to create beforehand.
This path sould be a POST to /lists